src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -284,156 +284,192 @@ public void FileInfo_Create_NullFileInfo()
+        [InlineData(FileMode.Create)]
+        [InlineData(FileMode.CreateNew)]
+        public void FileInfo_Create_FileSecurity_WriteModes_CreateFilesRights_AllowedCombo(FileMode mode) =>
+            FileInfo_Create_FileSecurity_Successful(mode, FileSystemRights.CreateFiles);
+
+        public static IEnumerable<object[]> AppendMode_UnexpectedReadRights_Data() =>
+            from writeRights in s_writableRights
+            from readRights in new[] {
+                FileSystemRights.ExecuteFile,
+                FileSystemRights.ReadAttributes, FileSystemRights.ReadData, FileSystemRights.ReadExtendedAttributes, FileSystemRights.ReadPermissions,
+                FileSystemRights.Read, // Contains ReadAttributes, ReadData, ReadExtendedAttributes, ReadPermissions
+                FileSystemRights.ReadAndExecute, // Contains Read and ExecuteFile
+            }
+            select new object[] { writeRights | readRights };
+
+        // Append is allowed if at least one write permission is provided
+        // But append is disallowed if any read rights are provided
+        [Theory]
+        [MemberData(nameof(AppendMode_UnexpectedReadRights_Data))]
+        public void FileInfo_Create_FileSecurity_AppendMode_UnexpectedReadRights(FileSystemRights rights) =>
+            FileInfo_Create_FileSecurity_ArgumentException(FileMode.Append, rights);
+
+        public static IEnumerable<object[]> AppendMode_OnlyWriteRights_Data() =>
+            from writeRights in s_writableRights
+            select new object[] { writeRights };
+
+        // Append succeeds if only write permissions were provided (no read permissions)
+        [Theory]
+        [MemberData(nameof(AppendMode_OnlyWriteRights_Data))]
+        public void FileInfo_Create_FileSecurity_AppendMode_OnlyWriteRights(FileSystemRights rights) =>
+            FileInfo_Create_FileSecurity_Successful(FileMode.Append, rights);
+
+        public static IEnumerable<object[]> WritableRights_Data() =>
+            from rights in s_writableRights
+            select new object[] { rights };
 
+        // Cannot truncate unless all write rights are provided
+        [Theory]
+        [MemberData(nameof(WritableRights_Data))]
+        public void FileInfo_Create_FileSecurity_TruncateMode_IncompleteWriteRights(FileSystemRights rights) =>
+            FileInfo_Create_FileSecurity_ArgumentException(FileMode.Truncate, rights);
+
+        // Truncate suceeds if all write rights are included
+        
+        // In .NET Framework, throws Could not find file (filename)
+        // In .NET, throws IOException: The parameter is incorrect (filename)
         [Fact]
-        public void FileInfo_Create_DefaultFileSecurity()
-        {
-            var security = new FileSecurity();
-            Verify_FileSecurity_CreateFile(security);
-        }
+        [ActiveIssue("Doesnt work in either .NET Framework or .NET, even though .NET Framework code expects this specific combo to succeed")]
```

Comment:
Most likely such combination is incorrect for the OS (Windows). Since we can't do anything about it, we should rather throw an exception that says that it's impossible (similarly to opening file for reading with Append mode) and have a test that verifies that.